examples/create_hdr_encoding_with_master_display_information_hls.go

```diff
@@ -0,0 +1,317 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bitmovin/bitmovin-go/bitmovin"
+	"github.com/bitmovin/bitmovin-go/bitmovintypes"
+	"github.com/bitmovin/bitmovin-go/models"
+	"github.com/bitmovin/bitmovin-go/services"
+)
+
+type H265CodecConfigDefinition struct {
+	width    *int64
+	height   *int64
+	crf      *float64
+	streamId *string
+	muxingId *string
+}
+
+const (
+	apiKey             = "YOUR API KEY"
+	inputHost          = "YOUR_HTTP_INPUT_HOST"
+	inputFilePath      = "/path/to/your/input/file.mov"
+	s3OutputAccessKey  = "YOUR_ACCESS_KEY"
+	s3OutputSecretKey  = "YOUR_SECRET_KEY"
+	s3OutputBucketName = "YOUR_BUCKET_NAME"
+	outputBasePath     = "golang_hls_example"
+)
+
+func main() {
+	// Creating Bitmovin object
+	bitmovin := bitmovin.NewBitmovin(apiKey, "https://api.bitmovin.com/v1/", 5)
+
+	videoEncodingProfiles := []*H265CodecConfigDefinition{
+		{width: intToPtr(4096), height: intToPtr(2160), crf: floatToPtr(19.0)},
+		{width: intToPtr(1920), height: intToPtr(1080), crf: floatToPtr(19.0)},
+		{width: intToPtr(1280), height: intToPtr(720), crf: floatToPtr(19.0)},
+	}
+
+	// Creating the HTTP Input
+	httpIS := services.NewHTTPInputService(bitmovin)
+	httpInput := &models.HTTPInput{
+		Host: stringToPtr(inputHost),
+	}
+	httpResp, err := httpIS.Create(httpInput)
+	errorHandler(httpResp.Status, err)
+
+	s3OS := services.NewS3OutputService(bitmovin)
+	s3Output := &models.S3Output{
+		AccessKey:  stringToPtr(s3OutputAccessKey),
+		SecretKey:  stringToPtr(s3OutputSecretKey),
+		BucketName: stringToPtr(s3OutputBucketName),
+	}
+	s3OutputResp, err := s3OS.Create(s3Output)
+	errorHandler(s3OutputResp.Status, err)
+
+	t := time.Now()
+	outputBasePath := outputBasePath + "/" + t.Format("2006-01-02-15-04-05")
```

Comment:
This is a good idea, so that no test encodes are overwritten by the user accidentally.